src/client.cpp

```diff
@@ -64,33 +67,268 @@ void set_client_timer(struct itimerval *timer) {
+/*  Display histogram to fit on terminal screen width (frequency rounded up) */
+void printHistogram(int binSize, std::map<int, int> &activeBins, int minValue, int maxValue) {
+    const int leftOutlierBinIndex = getLeftOutlierBinIndexReserved();
+    const int rightOutlierBinIndex = getRightOutlierBinIndexReserved();
+    const int lowerRange = s_user_params.histogram_lower_range;
+    const int upperRange = s_user_params.histogram_upper_range;
+    const std::string prefixToHistogramDisplay ("sockperf: ");
+    int maxFrequency = 0;
+    int currFrequency = 0;
+    int terminalWidth = 0;
+    int scalingUnit = 0;
+    int maxDisplayWidth = 0;
+    int maxStartBinDigits = getNumberOfDigits(maxValue - binSize);
+    int maxEndBinDigits = getNumberOfDigits(maxValue);
+    int whitespaceBeforeFrequencies = 2;
+    int outlierMessageWidth = 11;
+    int whitespaceAfterFrequencies = outlierMessageWidth + 1;
+    int columnWidthOfEndBinEdge = maxEndBinDigits + whitespaceBeforeFrequencies;
+    int columnWidthOfStartBinEdge = maxStartBinDigits;
+    int maxBinMessage = getNumberOfDigits(upperRange + binSize) + 1 + maxEndBinDigits; // align after longest message
+    std::map<int, int>::iterator itr;
+
+    // Scale to terminal
+#ifndef WIN32
+    struct winsize size;
+    ioctl(STDOUT_FILENO, TIOCGWINSZ, &size);
+    terminalWidth = size.ws_col;
+#else
+    CONSOLE_SCREEN_BUFFER_INFO csbi;
+    GetConsoleScreenBufferInfo(GetStdHandle(STD_OUTPUT_HANDLE),&csbi);
+    terminal_width = csbi.dwSize.X;
+#endif
+    for(itr = activeBins.begin(); itr != activeBins.end(); ++itr) {
+        currFrequency = itr->second;
+        if (currFrequency > maxFrequency) {
+            maxFrequency = currFrequency;
+        }
+    }
+    if(terminalWidth == 0) {
+        // This may happen when run without terminal, but we still want the file output to contain the display
+        int standardTerminalLength = 80;
+        scalingUnit = (maxFrequency + standardTerminalLength - 1)/standardTerminalLength;
+    } else {
+        maxDisplayWidth = terminalWidth - prefixToHistogramDisplay.length() - maxBinMessage - whitespaceAfterFrequencies
+            - whitespaceBeforeFrequencies;
+        scalingUnit = (maxFrequency + maxDisplayWidth - 1)/maxDisplayWidth; // round up
+    }
+
+    const std::string freqString = "frequency";
+    const std::string binsString = "bins";
+    int binsHeaderWidth = (maxStartBinDigits + maxEndBinDigits + 2)/2 + binsString.length()/2;
+    int frequencyHeaderWidth = (maxDisplayWidth)/2 + freqString.length()/2;
+    int startBinEdge = 0;
+    int endBinEdge = 0;
+    int frequency = 0;
+    int frequencyScaledDownCount = 0;
+
+    if (scalingUnit == 1) {
+        log_msg("[Histogram] Display to scale");
+    } else {
+        log_msg("[Histogram] Display scaled to fit on screen (Key: '#' = up to %d samples)", scalingUnit);
+    }
+    log_msg("%*s %*s", binsHeaderWidth, binsString.c_str(), frequencyHeaderWidth, freqString.c_str());
```

Comment:
Two whitespaces before frequencies as in line 203:
```suggestion
    log_msg("%*s  %*s", binsHeaderWidth, binsString.c_str(), frequencyHeaderWidth, freqString.c_str());
```